bake/hclparser/stdlib.go

```diff
@@ -254,6 +256,115 @@ func timestampFunc() function.Function {
+		Type: function.StaticReturnType(cty.Object(map[string]cty.Type{
+			"year":         cty.Number,
+			"year_day":     cty.Number,
+			"day":          cty.Number,
+			"month":        cty.Number,
+			"month_name":   cty.String,
+			"weekday":      cty.Number,
+			"weekday_name": cty.String,
+			"hour":         cty.Number,
+			"minute":       cty.Number,
+			"second":       cty.Number,
+			"unix":         cty.Number,
+			"iso_year":     cty.Number,
+			"iso_week":     cty.Number,
+		})),
+		Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+			ts := args[0].AsString()
+			rfc3339, err := time.Parse(time.RFC3339, ts)
+			if err != nil {
+				return cty.NilVal, err
+			}
+			isoYear, isoWeek := rfc3339.ISOWeek()
+			return cty.ObjectVal(map[string]cty.Value{
+				"year":         cty.NumberIntVal(int64(rfc3339.Year())),
+				"year_day":     cty.NumberIntVal(int64(rfc3339.YearDay())),
+				"day":          cty.NumberIntVal(int64(rfc3339.Day())),
+				"month":        cty.NumberIntVal(int64(rfc3339.Month())),
+				"month_name":   cty.StringVal(rfc3339.Month().String()),
+				"weekday":      cty.NumberIntVal(int64(rfc3339.Weekday())),
+				"weekday_name": cty.StringVal(rfc3339.Weekday().String()),
+				"hour":         cty.NumberIntVal(int64(rfc3339.Hour())),
+				"minute":       cty.NumberIntVal(int64(rfc3339.Minute())),
+				"second":       cty.NumberIntVal(int64(rfc3339.Second())),
+				"unix":         cty.NumberIntVal(int64(rfc3339.Unix())),
+				"iso_year":     cty.NumberIntVal(int64(isoYear)),
+				"iso_week":     cty.NumberIntVal(int64(isoWeek)),
+			}), nil
+		},
+	})
+}
+
+// unixtimestampParseFunc, given a unix timestamp integer, will parse and
+// return an object representation of that date and time
+//
+// This function is similar to the `unix_timestamp_parse` function in Terraform:
+// https://registry.terraform.io/providers/hashicorp/time/latest/docs/functions/unix_timestamp_parse
+func unixtimestampParseFunc() function.Function {
+	return function.New(&function.Spec{
+		Description: `Given a unix timestamp integer, will parse and return an object representation of that date and time. A unix timestamp is the number of seconds elapsed since January 1, 1970 UTC.`,
+		Params: []function.Parameter{
+			{
+				Name:        "unix_timestamp",
+				Description: "Unix Timestamp integer to parse",
+				Type:        cty.Number,
+			},
+		},
+		Type: function.StaticReturnType(cty.Object(map[string]cty.Type{
+			"year":         cty.Number,
+			"year_day":     cty.Number,
+			"day":          cty.Number,
+			"month":        cty.Number,
+			"month_name":   cty.String,
+			"weekday":      cty.Number,
+			"weekday_name": cty.String,
+			"hour":         cty.Number,
+			"minute":       cty.Number,
+			"second":       cty.Number,
+			"rfc3339":      cty.String,
+			"iso_year":     cty.Number,
+			"iso_week":     cty.Number,
+		})),
+		Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+			ts, _ := args[0].AsBigFloat().Int64()
```

Comment:
see above

Reply:
You might want to handle the error.

Reply:
> You might want to also handle the error.

`.Int64()` does not return an error as second value but accuracy: https://pkg.go.dev/math/big#Float.Int64